homeassistant/components/ring/binary_sensor.py

```diff
@@ -53,70 +69,88 @@ async def async_setup_entry(
 ) -> None:
     """Set up the Ring binary sensors from a config entry."""
     ring_data = entry.runtime_data
+    listen_coordinator = ring_data.listen_coordinator
 
-    entities = [
-        RingBinarySensor(
-            ring_data.api,
-            device,
-            ring_data.notifications_coordinator,
-            description,
-        )
+    async_add_entities(
+        RingBinarySensor(device, listen_coordinator, description)
         for description in BINARY_SENSOR_TYPES
         for device in ring_data.devices.all_devices
-        if description.exists_fn(device)
-    ]
-
-    async_add_entities(entities)
+        if device.has_capability(description.capability)
+        and async_check_create_deprecated(
+            hass,
+            Platform.BINARY_SENSOR,
+            f"{device.id}-{description.key}",
+            description,
+        )
+    )
 
 
 class RingBinarySensor(
-    RingBaseEntity[RingNotificationsCoordinator], BinarySensorEntity
+    RingBaseEntity[RingListenCoordinator, RingDeviceT], BinarySensorEntity
 ):
     """A binary sensor implementation for Ring device."""
 
     _active_alert: RingEvent | None = None
-    entity_description: RingBinarySensorEntityDescription
+    RingBinarySensorEntityDescription[RingDeviceT]
 
     def __init__(
         self,
-        ring: Ring,
-        device: RingGeneric,
-        coordinator: RingNotificationsCoordinator,
-        description: RingBinarySensorEntityDescription,
+        device: RingDeviceT,
+        coordinator: RingListenCoordinator,
+        description: RingBinarySensorEntityDescription[RingDeviceT],
     ) -> None:
-        """Initialize a sensor for Ring device."""
+        """Initialize a binary sensor for Ring device."""
         super().__init__(
             device,
             coordinator,
         )
         self.entity_description = description
-        self._ring = ring
         self._attr_unique_id = f"{device.id}-{description.key}"
-        self._update_alert()
+        self._attr_is_on = False
+        self._active_alert: RingEvent | None = None
+        self._cancel_callback: TimerHandle | None = None
 
     @callback
-    def _handle_coordinator_update(self, _: Any = None) -> None:
-        """Call update method."""
-        self._update_alert()
-        super()._handle_coordinator_update()
+    def _async_handle_event(self, alert: RingEvent) -> None:
+        """Handle the event."""
+        self._attr_is_on = True
+        self._active_alert = alert
+        loop = self.hass.loop
+        when = loop.time() + alert.expires_in
+        if self._cancel_callback:
+            self._cancel_callback.cancel()
+        self._cancel_callback = loop.call_at(when, self._async_cancel_event)
```

Comment:
Please use the event helpers here `homeassistant.helpers.event`

Reply:
Done